files/usr/share/cinnamon/applets/menu@cinnamon.org/applet.js

```diff
@@ -3072,18 +3075,17 @@ class CinnamonMenuApplet extends Applet.TextIconApplet {
                 });
             }
         } else if (apps) {
-            for (let i = 0; i < this._applicationsButtons.length; i++) {
-                let button = this._applicationsButtons[i];
+            this._applicationsButtons.forEach(button => {
```

Comment:
`Array.prototype.forEach` is slower than for loops.

You can try [these](https://jsperf.com/for-vs-foreach/75) [tests](https://jsperf.com/loops/33) in Firefox, which uses the same JS engine as Cinnamon.

An alternative is the the `each` function in misc/util, which wraps a for loop.

```js
Util.each(this._applicationsButtons, (button) => {...})
```

Reply:
forEach is used all over the place in this file. We are talking about the menu with, say, max. 1000 entries (and that would be a lot). So IMHO it is fine to use forEach here.